Divide by Zero Exception fixed; Added Log2(x) feature

SRvSaha opened the pull request.

SRvSaha committed:
Divide by Zero Issue Fixed

trishnaguha commented on main.py:
```diff
@@ -2,122 +2,181 @@
+
+
+def add(a, b):
+    val = a + b
```
Can you move the docstring inside method? and same for rest of the methods?

SRvSaha replied:
Well, I can do that. Will be updating the asked changes in some time.

trishnaguha commented on main.py:
```diff
@@ -2,122 +2,181 @@
-    parser = argparse.ArgumentParser(description = "A Command Line Calculator", epilog = "Enter two numbers for addition,subtraction and so on....\n Enter only one number for percentage calculation,factorial and so on....")
-	
+    parser = argparse.ArgumentParser(description="A Command Line Calculator",
+                                     epilog="Enter two numbers for addition,subtraction and so on....\n Enter only one number for percentage calculation,factorial and so on....")
```
It would be nice if we can follow PEP8 rule. Could you break the text in couple of lines?

SRvSaha replied:
Actually, I did Auto PEP8 formatting using Anaconda plugin in Sublime Text. 

Anyways, I will do the changes. :) Thanks for giving me an opportunity

trishnaguha commented:
Otherwise Looks good to me.

SRvSaha commented:
![image](https://cloud.githubusercontent.com/assets/13312352/20102979/15e6d4e4-a5ee-11e6-860b-bda01c02b71b.png)

I think there's something wrong going on behind the scene.
As per the code there should have been a new line after so on... And the Enter only one num bla bla should come in new line. But why is this "\n" not working ? Should I keep it the way it is any fix the issue of "line too long " according to PEP8 or should I work on the "\n" thing as well??
<code>parser = argparse.ArgumentParser(description="A Command Line Calculator",
                                     epilog="Enter two numbers for addition,subtraction and so on....\n Enter only one number for percentage calculation,factorial and so on....")
</code>
"

SRvSaha committed:
Fixed PEP8 errors

SRvSaha commented:
Glad to fix all the issues pointed out by you and the ones I found that needed to be fixed.

Fixed Issues :
1. PEP8 Guideline-based DOCSTRING ERROR and LINE TOO LONG ERROR.
2. Added formatter class for epilog in argument parser for better readability & fixed the newline issue.
3. The percentage method is modified a bit so that the division is more generic(support of Python2 in Testing).

Hoping for a merge for my first contribution to Open Source :)

SRvSaha committed:
Added feature : log2(x)

SRvSaha committed:
Added Test for log2(x) ; Fixed PEP8 errors

SRvSaha committed:
Added Logarithm2(x)

SRvSaha commented:
Changed made :
1. Added feature of log2(x) which is very important since most of the times we deal with log2(x) in Computer Science.
2. Updated Readme file with the logarithm2(x)
3. Fixed all PEP8 errors (except docstring missing in test_main.py) in the codebase.

trishnaguha commented:
Looks good! Thanks @SRvSaha .
